app/ide-desktop/lib/dashboard/src/data/dataLinkSchema.json

```diff
@@ -88,32 +89,35 @@
       "type": "object",
       "properties": {
         "type": { "title": "Type", "const": "S3", "type": "string" },
+        "libraryName": { "const": "Standard.AWS" },
         "uri": {
           "title": "URI",
           "description": "Must start with \"s3://\".",
           "type": "string",
-          "pattern": "^s3://[\\w.~-]+/[/\\w.~-]+$"
+          "pattern": "^s3://[a-z0-9.-]{3,63}/.{1,1024}$"
```

Comment:
The S3 bucket name can only contain _lowercase letters_, digits, `.` and `-`. It has to be at least 3 characters long and at most 63 characters long. 

There are a few more restrictions - e.g. it cannot start with `-`, it cannot contain double `--` etc. But I thought that just checking the simple ones is enough - it's just a basic sanity check - even if the bucket name is valid, the bucket could not exist / be not accessible anyway, so this does not need to be comprehensive.

The bucket key can be arbitrary, but it has a limit of 1024 bytes, so the length limit is a heuristic of that.